test/store_test.dart

```diff
@@ -65,58 +65,70 @@ void main() {
     });
 
     test('should trigger in response to an action', () {
-      Action _action = new Action();
-      store.triggerOnAction(_action);
-      store.listen((Store payload) => expectAsync((payload) {
-            expect(payload, equals(store));
-          }));
-      _action();
+      var c = new Completer();
+      var action = new Action();
+      store.triggerOnAction(action);
+      store.listen(expectAsync((payload) {
+        c.complete();
+        expect(payload, equals(store));
+      }));
+      action();
+      return c.future;
     });
 
     test(
         'should execute a given method and then trigger in response to an action',
         () {
-      Action _action = new Action();
-      bool methodCalled = false;
+      var c = new Completer();
+      var action = new Action();
+      var methodCalled = false;
       syncCallback(_) {
         methodCalled = true;
       }
-      store.triggerOnAction(_action, syncCallback);
-      store.listen((Store payload) => expectAsync((payload) {
-            expect(payload, equals(store));
-            expect(methodCalled, equals(true));
-          }));
-      _action();
+      store.triggerOnAction(action, syncCallback);
+      store.listen(expectAsync((payload) {
+        c.complete();
+        expect(payload, equals(store));
+        expect(methodCalled, isTrue);
+      }));
+      action();
+      return c.future;
     });
 
     test(
         'should execute a given async method and then trigger in response to an action',
         () {
-      Action _action = new Action();
-      bool afterTimer = false;
+      var c = new Completer();
```

Comment:
Consider using `final` instead of `var`. This allows the Dart analyzer to warn you if you try to overwrite the variable with an incompatible type. For example, after your statement above, this would work properly:
```
c = 5;
```
If you use `final`, the above line would be flagged by the analyzer.